snuba/datasets/entities/generic_metrics.py

```diff
@@ -0,0 +1,63 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import (
+    AggregateFunction,
+    Column,
+    ColumnSet,
+    DateTime,
+    Nested,
+    UInt,
+)
+from snuba.clickhouse.translators.snuba.mappers import (
+    FunctionNameMapper,
+    SubscriptableMapper,
+)
+from snuba.clickhouse.translators.snuba.mapping import TranslationMappers
+from snuba.datasets.entity import Entity
+from snuba.datasets.plans.single_storage import SingleStorageQueryPlanBuilder
+from snuba.datasets.storages.generic_metrics import sets_bucket_storage, sets_storage
+from snuba.pipeline.simple_pipeline import SimplePipelineBuilder
+from snuba.query.processors import QueryProcessor
+from snuba.query.validation.validators import EntityRequiredColumnValidator
+
+
+class GenericMetricsSetsEntity(Entity):
```

Comment:
How are you going to manage the granularity in this dataset ?
https://github.com/getsentry/snuba/blob/master/snuba/datasets/entities/metrics.py#L112

The product is used to provide the granularity in seconds. Now granularity are discrete values instead. Are we going to solve the discrepancy here or in the product ?

Reply:
I was planning to hide this detail from product and have them send granularity in seconds (https://getsentry.atlassian.net/browse/SNS-1480)

Reply:
I think we should think twice before doing that as it is exactly against the idea of removing the logical data model from Snuba. 
The transformation from seconds to the specific granularity enum value is arguably a product concern, that would fit in the snuba logical data model as an entity query processor today, but which we should move away from.

Any chance to move this into Sentry ?

Reply:
If it is a big deal we can prioritize deploying this first.

Reply:
yeah, I'm going to deploy this as-is and we can finalize the granularity approach in the next few days

Reply:
Late to the party, but +1 on making this a product concern instead of a hidden Snuba feature.